.circleci/config.yml

```diff
@@ -1,90 +1,114 @@
 version: 2.0
 
 aliases:
-  - &docker_agent datadog/docker-dd-agent
-  - agent_env: &agent_env
-    - DD_APM_ENABLED=true
-    - DD_BIND_HOST=0.0.0.0
-    - DD_API_KEY=invalid_key_but_this_is_fine
-  - &load_composer_deps
+  - &CACHE_COMPOSER_KEY
     key: 'betav1-composer-deps-{{ .Environment.CIRCLE_JOB }}-{{ checksum "composer.json" }}'
-  - &composer_install_deps
-    name: Composer installing dependencies
-    command: composer install -n
-  - &save_composer_deps
-    key: 'betav1-composer-deps-{{ .Environment.CIRCLE_JOB }}-{{ checksum "composer.json" }}'
-    paths:
-      - vendor/
-  - &test_results_path
-    path: test-results
-  - &php_test_defaults
+
+  - &CACHE_NPM_KEY
+    key: 'betav1-lint-deps-{{ checksum "composer.json" }}'
+
+  - &IMAGE_DOCKER_DD_AGENT
+    image: datadog/docker-dd-agent
+    environment:
+      - DD_APM_ENABLED=true
+      - DD_BIND_HOST=0.0.0.0
+      - DD_API_KEY=invalid_key_but_this_is_fine
+
+  - &STEP_COMPOSER_CACHE_RESTORE
+    restore_cache:
+      <<: *CACHE_COMPOSER_KEY
+
+  - &STEP_COMPOSER_CACHE_SAVE
+    save_cache:
+      <<: *CACHE_COMPOSER_KEY
+      paths:
+        - vendor/
+
+  - &STEP_COMPOSER_INSTALL
+    run:
+      name: Installing dependencies with composer
+      command: composer install -n
+
+  - &STEP_STORE_ARTIFACTS
+    store_artifacts:
+      path: test-results
+
+  - &STEP_STORE_TEST_RESULTS
+    store_test_results:
+      path: test-results
+
+  - &JOB_PHP_DEFAULTS
     working_directory: ~/datadog
     steps:
       - checkout
-      - restore_cache: *load_composer_deps
-      - run: *composer_install_deps
-      - save_cache: *save_composer_deps
+      - <<: *STEP_COMPOSER_CACHE_RESTORE
+      - <<: *STEP_COMPOSER_INSTALL
+      - <<: *STEP_COMPOSER_CACHE_SAVE
       - run:
-          name: 'Waiting for Dockerized agent'
+          name: Waiting for Dockerized agent
           command: dockerize -wait tcp://127.0.0.1:8126 -timeout 1m
-      - run:
-          name: Run unit tests
-          command: composer test-unit -- --log-junit test-results/php-unit/results.xml
       - run:
           name: Run integration tests
           command: composer test-integration -- --log-junit test-results/php-integration/results.xml
-      - store_test_results: *test_results_path
-      - store_artifacts: *test_results_path
+      - run:
```

Comment:
Was moving the unit tests after the integration tests intentional? Right now all of the tests are quite fast, but usually unit tests are faster, so I'd imagine we want them to run first so failures are faster (could probably also run them before starting the agent).

Reply:
@chuck 

No, that wasn’t intended. I’ll fix and let you know!